packages/client/components/BottomControlBarReadyButton.tsx

```diff
@@ -0,0 +1,183 @@
+import graphql from 'babel-plugin-relay/macro'
+import React, {ReactNode, useEffect, useState} from 'react'
+import {useFragment} from 'react-relay'
+import {BottomControlBarReadyButton_meeting$key} from '~/__generated__/BottomControlBarReadyButton_meeting.graphql'
+import useAtmosphere from '../hooks/useAtmosphere'
+import isDemoRoute from '../utils/isDemoRoute'
+
+const shake = keyframes`
+  20%, 40%, 60% {
+    padding: 4px 4px 8px;
+    overflow: hidden;
+    background: #a06bd6;
+    color: white;
+    border-radius: 4px;
+    box-shadow: 0px 9px 12px rgba(68, 66, 88, 0.14), 0px 3px 16px rgba(68, 66, 88, 0.12), 0px 5px 6px rgba(68, 66, 88, 0.2);
+  }
+  0% {
+    transform: rotate(0deg);
+  }
+  20% {
+    transform: rotate(-7deg);
+  }
+  40% {
+    transform: rotate(4deg);
+  }
+  60% {
+    transform: rotate(-7deg);
+  }
+  100% {
+    transform: rotate(0deg);
+  }
+`
+
+const BottomNavReadyButton = styled('div')<{shouldAnimate?: boolean; delay: number}>(
+  ({shouldAnimate, delay}) => {
+    return {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      animation: shouldAnimate ? `1s ease-in-out ${delay}s 3 ${shake}` : undefined
+    }
+  }
+)
+
+interface BottomControlBarReadyButton {
+  meetingRef: BottomControlBarReadyButton_meeting$key
+  children: ReactNode
+}
+
+const BottomControlBarReadyButton = (props: BottomControlBarReadyButton) => {
+  const {meetingRef, children} = props
+  const meeting = useFragment(
+    graphql`
+      fragment BottomControlBarReadyButton_meeting on NewMeeting {
+        localPhase {
+          phaseType
+        }
+        localStage {
+          readyCount
+          isViewerReady
+        }
+        meetingMembers {
+          id
+        }
+        ... on RetrospectiveMeeting {
+          endedAt
+          facilitatorUserId
+          localPhase {
+            ... on ReflectPhase {
+              reflectPrompts {
+                id
+                editorIds
+              }
+            }
+          }
+          reflectionGroups {
+            id
+            reflections {
+              id
+              isEditing
+              isDropping
+              isViewerDragging
+            }
+          }
+          viewerMeetingMember {
+            votesRemaining
+          }
+          votesRemaining
+        }
+      }
+    `,
+    meetingRef
+  )
+  const {
+    endedAt,
+    facilitatorUserId,
+    localPhase,
+    localStage,
+    meetingMembers,
+    reflectionGroups,
+    viewerMeetingMember,
+    votesRemaining
+  } = meeting
+  const [delaySeconds, setDelaySeconds] = useState(2)
+  const [shouldAnimate, setShouldAnimate] = useState(false)
+  const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
+
+  useEffect(() => {
+    // only enable for facilitatorUser
+    const isFacilitator = viewerId === facilitatorUserId && !endedAt
+    if (!isFacilitator) return
+
+    switch (localPhase.phaseType) {
+      case 'reflect':
```

Comment:
+1 I think it would be more readable if you would split it up into multiple `useEffect` blocks:
```ts
useEffect(() => {
    // only enable for facilitatorUser
    const isFacilitator = viewerId === facilitatorUserId && !endedAt
    if (!isFacilitator) return
    if (localPhase.phaseType !== 'reflect') return

      const hasNoReflection = reflectionGroups?.every((group) => group.reflections?.length === 0)
      if (hasNoReflection) return

      const reflectPrompts = localPhase.reflectPrompts!
      const beforeDemoStart =
        isDemoRoute() && reflectPrompts.every(({editorIds}) => !Array.isArray(editorIds))
      if (beforeDemoStart) return

      const isNotEditing = reflectPrompts.every(({editorIds}) => {
        return editorIds === undefined || (Array.isArray(editorIds) && editorIds.length === 0)
      })

      setDelaySeconds(30)
      setShouldAnimate(isNotEditing)
    }, [localPhase.phaseType === 'reflect', reflectionGroups, editorIds])
```

Reply:
This is good practice, as #TIL. Just want to confirm, is the `=== condition` inside the deps of `useEffect` a generic pattern? Is there a reference for this? 😁

```ts
[localPhase.phaseType === 'reflect', reflectionGroups, editorIds])
```

Reply:
I don't think this is common practice. You need to match what's used inside the block, so `localPhase.phaseType` would be sufficient, it would just run the block more often than necessary.

Reply:
I took a look and found this
https://github.com/ParabolInc/parabol/blob/7bb4774f8ec3edc9022d0f7d26f7874865a579ae/packages/client/components/GroupingKanban.tsx#L73